### ISSUE TITLE
Building on OSX 10.14 for SIMULATOR64 (i386) fails

### ISSUE THREAD

**Issue by kohenchia** [pytorch/QNNPACK#5]
The `make` process tries to compile ARM code (`src/arm/match/init.c`) when building on OSX 10.14 for SIMULATOR64 (i386).

Command:
```
$ cmake .. -DCMAKE_TOOLCHAIN_FILE=../../ios-cmake/ios.toolchain.cmake -DIOS_PLATFORM=SIMULATOR64
$ make
```

Error:
```
Scanning dependencies of target cpuinfo
[ 29%] Building C object deps/cpuinfo/CMakeFiles/cpuinfo.dir/src/init.c.o
[ 30%] Building C object deps/cpuinfo/CMakeFiles/cpuinfo.dir/src/api.c.o
[ 32%] Building C object deps/cpuinfo/CMakeFiles/cpuinfo.dir/src/arm/uarch.c.o
[ 33%] Building C object deps/cpuinfo/CMakeFiles/cpuinfo.dir/src/arm/cache.c.o
[ 35%] Building C object deps/cpuinfo/CMakeFiles/cpuinfo.dir/src/arm/mach/init.c.o
/Users/kohenchia/github/QNNPACK/deps/cpuinfo/src/arm/mach/init.c:19:24: error: redefinition of 'cpuinfo_isa' with a different type:
      'struct cpuinfo_arm_isa' vs 'struct cpuinfo_x86_isa'
struct cpuinfo_arm_isa cpuinfo_isa = {
                       ^
/Users/kohenchia/github/QNNPACK/deps/cpuinfo/include/cpuinfo.h:683:32: note: previous declaration is here
        extern struct cpuinfo_x86_isa cpuinfo_isa;
                                      ^
/Users/kohenchia/github/QNNPACK/deps/cpuinfo/src/arm/mach/init.c:287:16: error: no member named 'sha1' in 'struct cpuinfo_x86_isa'; did
      you mean 'sha'?
                        cpuinfo_isa.sha1 = true;
                                    ^~~~
                                    sha
/Users/kohenchia/github/QNNPACK/deps/cpuinfo/include/cpuinfo.h:674:8: note: 'sha' declared here
                bool sha;
                     ^
/Users/kohenchia/github/QNNPACK/deps/cpuinfo/src/arm/mach/init.c:288:16: error: no member named 'sha2' in 'struct cpuinfo_x86_isa'; did
      you mean 'sha'?
                        cpuinfo_isa.sha2 = true;
                                    ^~~~
                                    sha
/Users/kohenchia/github/QNNPACK/deps/cpuinfo/include/cpuinfo.h:674:8: note: 'sha' declared here
                bool sha;
                     ^
/Users/kohenchia/github/QNNPACK/deps/cpuinfo/src/arm/mach/init.c:289:16: error: no member named 'pmull' in 'struct cpuinfo_x86_isa'
                        cpuinfo_isa.pmull = true;
                        ~~~~~~~~~~~ ^
/Users/kohenchia/github/QNNPACK/deps/cpuinfo/src/arm/mach/init.c:290:16: error: no member named 'crc32' in 'struct cpuinfo_x86_isa'
                        cpuinfo_isa.crc32 = true;
                        ~~~~~~~~~~~ ^
5 errors generated.
make[2]: *** [deps/cpuinfo/CMakeFiles/cpuinfo.dir/src/arm/mach/init.c.o] Error 1
make[1]: *** [deps/cpuinfo/CMakeFiles/cpuinfo.dir/all] Error 2
make: *** [all] Error 2
```

This is because `CMAKE_SYSTEM_PROCESSOR` is empty on my machine. It is sort of a known bug with CMake: https://public.kitware.com/Bug/view.php?id=9065. The if clause on `deps/cpuinfo/CMakeLists.txt` fails, and proceeds to include `src/arm/mach/init.c`. 

In order to fix it, I had to manually set `CMAKE_SYSTEM_PROCESSOR` to ~~`i386`~~ `x86_64`. (Edit: Setting to `i386` didn't work, which is a separate problem.)

**Comment by Maratyszcza** [comment]
cpuinfo assumed that iOS is always ARM, which isn't true to simulators. Issue is fixed in pytorch/cpuinfo@e901c0790eee35ad6a947713e71a579bddef825e

**Comment by Maratyszcza** [comment]
SIMULATOR64 (and other x86+iOS builds) are fixed in 9ab804eef0e605cd9794ae78ca1058306ff240b6
